blog/2019-03-28-debezium-at-pipedrive.adoc

```diff
@@ -0,0 +1,225 @@
+
+So what we’ve done so far (to make it perform acceptable level) is apply strict table whitelist for tracking 25% of the most critical tables. For large history topic configured connector to only contain whitelisted table DDLs. Also increasing java heap size to ~2GB has so far proven reliable.
+
+[source,json]
+----
+database.history.store.only.monitored.tables.ddl: "true",
+database.history.skip.unparseable.ddl": "true",
+----
+
+=== Snapshots without locking
+You can configure connector to use locking.mode=’none’, but it’s dangerous as ALTERS running during the snapshot will not be reflected in the Debezium in-memory database schema. You will learn about it only when the first data change event arrives for that table, which could be weeks after snapshot. So far the fix that worked has been to run schema recovery asap.
+
+== Lessons learned
+*Kafka connect in distributed mode is unstable*. Kafka-connect cluster doesn’t play well with large connectors count and slow task startup time - one connector restart can cause rebalance event, which means one repeatedly failing MySQL connector can halt whole cluster.
+
+*Manual Debezium offsets adjusting has saved us many times* - storing offsets in external kafka topic allowed us to easily adjust connector’s last offset by producing valid json message and restarting container. With active-passive setup unfortunate server failure can introduce new gaps in gtid ranges or slave or debezium itself can become ahead of new master on particular gtid channel. Fixing this without reseting connector needs manual intervention to last offset.
+
+*Schema / ddl parser problems really hurt*. There’s no easy way out as the alter breaking Debezium ddl parser can come from any database/table (whitelist doesn't apply) and is also written into history topic. Even if you push the debezium offset manually over the failing event you’ll crash again while reading from the history topic. Only thing that works is running schema recovery, but this only when no other alters have been executed.
+
+But once you get it running stable you’ll see uptime for months.
+
+== Future
+
+Change data capture remains a huge part of our system and has proven to produce consistent and reliable data. Setting up and operations hasn’t been the smoothest, but it does work delivering initial goal of zero data loss. We have production services depending on it every day and more being built.
```

Comment:
Maybe you could add a sentence or two, stating that you're working with us (Debezium team) to further improve problems you've ran into? You already did apply a nice set of changes, would be cool to highlight that, too.